lint-gitlab-ci.sh

```diff
@@ -1,27 +1,46 @@
 #!/usr/bin/env bash
 set -e
 
-GITLAB_SERVER_URL=${GITLAB_SERVER_URL:-"gitlab.com"}
+# GITLAB_SERVER_URL=${GITLAB_SERVER_URL:-"gitlab.com"}
+# ## get gitlab push remote
+GITLAB_REMOTE=$(git remote -v | grep push | awk '{print $2}')
+# ## remove useless parts
+# ## git urls (everything bevore @)
+GITLAB_REMOTE=${GITLAB_REMOTE#*@}
+# ## https urls (everythin before //)
+GITLAB_REMOTE=${GITLAB_REMOTE#*//}
+# ## .git ending
+GITLAB_REMOTE=${GITLAB_REMOTE%.git}
+
+# ## get the url
+GITLAB_SERVER_URL=$(echo "${GITLAB_REMOTE:?}" | cut -d':' -f1)
+# ## get the project name and URL encode
+GITLAB_PROJECT=$(echo "${GITLAB_REMOTE:?}" | cut -d':' -f2 | sed 's#/#%2F#g' )
+# ## get project ID
+# GITLAB_PROJECT_ID=$(curl -s --header "PRIVATE-TOKEN: ${GITLAB_TOKEN:?}" "https://${GITLAB_SERVER_URL:?}/api/v4/projects/${GITLAB_PROJECT:?}" | jq '.id')
```

Comment:
This should not be commented out, right?

Reply:
I first used the project ID until I realized, that `:id` can be either the ID or an URL encoded path to the project (see e.g. <https://docs.gitlab.com/ee/api/projects.html#get-single-project>). Using the URL encoded path, which in this case essentially means `/` by `%2F`,  makes the additional API which would be required to get the actual ID, redundant.

So `GITLAB_PROJECT_ID` is not needed. Using `GITLAB_PROJECT` is more efficient (no extra call to the GitLab API).

Reply:
ah I see. I tested it with a repository that I recently moved to another group, but did not update the git remotes. I guess that is something we could document on the README.

However, nice job!